usaspending_api/search/v2/views/spending_by_category.py

```diff
@@ -218,6 +220,43 @@ def funding_agency(self) -> list:
             row["id"] = fetch_agency_tier_id_by_agency(row["name"], self.category == "funding_subagency")
         return results
 
+    @staticmethod
+    def _get_recipient_id(row):
+        """
+        In the recipient_profile table there is a 1 to 1 relationship between hashes and DUNS
+        (recipient_unique_id) and the hashes+duns match exactly between recipient_profile and
+        recipient_lookup where there are matches.  Grab the level from recipient_profile by
+        hash if we have one or by DUNS if we have one of those.
+        """
+        if "recipient_hash" in row:
+            profile_filter = {"recipient_hash": row["recipient_hash"]}
+        elif "recipient_unique_id" in row:
+            profile_filter = {"recipient_unique_id": row["recipient_unique_id"]}
+        else:
+            raise RuntimeError(
+                "Attempted to lookup recipient profile using a queryset that contains neither "
+                "'recipient_hash' nor 'recipient_unique_id'"
+            )
+
+        profile = (
+            RecipientProfile.objects.filter(**profile_filter)
+            .exclude(recipient_name__in=SPECIAL_CASES)
+            .annotate(sort_order=Case(
+                When(recipient_level="C", then=Value(0)),
```

Comment:
Took me a minute to understand this case statement is for sorting.

Reply:
So the name `sort_order` didn't  make it clear enough?  :-p

Reply:
It's been a long week

Reply:
lol ... I feel ya brudda ... coming up on 17 hours so far today.